sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1269,10 +1286,14 @@ ProgramManager::ProgramPtr ProgramManager::build(
   static bool ForceLink = ForceLinkEnv && (*ForceLinkEnv == '1');
 
   const PluginPtr &Plugin = Context->getPlugin();
+  std::string CompileOptionsString(CompileOptions);
+  std::string LinkOptionsString(LinkOptions);
```

Comment:
Why do we need to create these and other strings within this function?

Reply:
The function piProgramCompile expects a null-terminated string and if we were to use string_view::data() it would return a pointer to a buffer that is not necessarily null-terminated.

Reply:
When can we get a non-null-terminated string there?